### DISCUSSION TITLE
Failed Workflow not available in the UI

### DISCUSSION THREAD

**Discussion by kathir0301** [Netflix/conductor#3277] [Q&A]
When the workflows are failed , its triggering the failure workflow but the parent workflow is not available in the UI due to which not able to track the parent workflow . Its happening when the following parameters are included in the workflow

**"failureWorkflow": "exception-flow",
 "workflowStatusListenerEnabled": true,**

Below is the workflow definition for reference , 

{
  "name": "terminate-test",
  "description": "Test Terminate Workflow",
  "version": 1,
  "tasks": [
    {
      "name": "set_workflow_id",
      "taskReferenceName": "set_workflowId",
      "inputParameters": {
        "workflowId": "${workflow.workflowId}"
      },
      "type": "SET_VARIABLE",
      "decisionCases": {},
      "defaultCase": [],
      "forkTasks": [],
      "startDelay": 0,
      "joinOn": [],
      "optional": false,
      "defaultExclusiveJoinTask": [],
      "asyncComplete": false,
      "loopOver": []
    },
    {
      "name": "terminate",
      "taskReferenceName": "terminate0",
      "inputParameters": {
        "terminationStatus": "FAILED",
        "workflowOutput": {
          "output": "This is a workflow output"
        },
        "terminationReason": "Custom terminationReason Reason"
      },
      "type": "TERMINATE",
      "decisionCases": {},
      "defaultCase": [],
      "forkTasks": [],
      "startDelay": 0,
      "joinOn": [],
      "optional": false,
      "defaultExclusiveJoinTask": [],
      "asyncComplete": false,
      "loopOver": []
    }
  ],
  "inputParameters": [
    "company_id",
    "document_id",
    "transaction_id",
    "batch_id"
  ],
  "outputParameters": {},
  "failureWorkflow": "exception-flow",
  "workflowStatusListenerEnabled": true,
  "schemaVersion": 2,
  "restartable": true,
  "ownerEmail": "test@conductor.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 0,
  "variables": {},
  "inputTemplate": {}
}

When this workflow is triggered exception flow is available in the UI but the terminate-test is missing in the UI .

Please provide your thoughts if its bug or anything missing in the workflow .

Any help is appreciated .Thanks in advance

**Comment by apanicker-nflx** [comment]
I used the workflow definition above and was able to view the parent and failure workflow in the UI. Could you check if you have an archival process that cleans up workflows in terminal state that may be cleaning up the parent workflow?

**Reply by kathir0301** [reply]
@apanicker-nflx We have enabled the archival process but able to see other failed jobs(parent jobs) in the UI,
 this is happening only when failure workflow is added to workflow .

When the parent fails it  triggers the failure workflow , able to see failure workflow in UI but not able to view parent workflow in the UI, due to this not able to troubleshoot the parent workflow failure.

**Reply by apanicker-nflx** [reply]
I am not able to reproduce this behavior using your workflows and the setup that you described above.
Please check for failure logs at your persistence layer or on the archival layer.

**Reply by kathir0301** [reply]
I looked in to the logs and persistence layer .Here is couple of observations . 
1. Parent workflow which was failing is archived(removed ) from postgres database .
2. And I'm able to query  the parent workflow details in the elastic search . 
3. Looks like UI is fetching the details only from the Postgres database. and its failing to fetch from elastic sear
4. From the logs , its failing at ElasticSearchRestDAOV6 class get method even though the workflow is available in the elastic search .

Replicate the scenario:
1. Enable the archival for the workflows
2. create the parent workflow as defined above .
3. Create the corresponding failure flow .
4. Upon running the workflow , parent workflow would be missing in the UI and only the failure workflow will be available in the UI.

Please let me know if you need  any other details

**Reply by kathir0301** [reply]
@apanicker-nflx Upon debugging noticed that on the failure flow after the workflow is archived , the workflow is updated again at 
com.netflix.conductor.core.execution.WorkflowExecutor.java (lineno :1009) .

This is resetting workflow details from the elastic search 
and the rawjson field which is required for the UI is removed .And as a result parent worflow is not available in the UI.
Can this this update statement which I mentioned above can be removed?

**Reply by aryan-gupta-acko** [reply]
Hi @kathir0301, Can you please share an update? How did you go ahead with this, did you comment out the `update` method?
